src/odf/dataset/src/entities/dataset.rs

```diff
@@ -23,6 +23,16 @@ use crate::{AppendError, BlockRef, DatasetSummary, MetadataChain};
 
 #[async_trait]
 pub trait Dataset: Send + Sync {
+    /// Detaches this dataset from any transaction references
```

Comment:
Same here about the link to a tech debt issue

Reply:
Same here, I don't like calling it a tech debt until we have a better suggestion.